docs/code_overview.md

```diff
@@ -1,52 +1,51 @@
 
-## **Diagram**
-![what](./resources/program_structure2.png)
+The code is split into two Github repositories: urc-2023 and mission-control. Keep in mind that the mission control breaks the code design a little as
+it is not running on the rover computer but rather over the network on your laptop! The mission control should never represent the state of the rover. 
 
-## **Components of the CodeBase**
-### **CAN Related**
+## The Core Idea
 
-### **`can_moteus`**
-[Docs link](./can_moteus.md)
+The code is split into three parts:
 
-This is a **ROS Node** that is responsible for communication between the [**Moteus motor controllers**](https://github.com/mjbots/moteus) (via a FDCAN/CAN Bus) and **ROS** that the rest of the codebase is written in.
+1. **Low-level code** -> This deals with the funky hardware communication such as USB, CAN bus, Wi-Fi, etc. It prepares publishers/subscribers for the interface to use
 
-### **`robot_interface`**
-This is responsible for converting human units into Moteus units that the Moteus controllers can understand. This is **not** a ROS Node, and instead will be an object that **takes in** a ROS Node to subscribe to data published by `can_moteus`. There are a long list of methods that correspond to the functionality of the rover; i.e. `setBottomArmMotorAngle(123)` would set the angle of the bottom motor of the arm to 123 degrees.
+2. **The interface** -> This deals making it easy to interact with the rover by exposing functions such as `runMotor()` or `getMotorState()`
 
-### **Robot Functionality**
+3. **High-level code** -> This implements the subsystems found on the rover. It directly talks to the interface in order to get the rover to physical do things. 
+Notice how the high-level code does not care about the hardware, thus making it easier to focus on the **algorithms** rather than fighting the hardware.
 
-### **`antenna`**
-This is responsible for always pointing the robot antenna towards the base station's antenna to secure the strongest WiFi signal.
+![Code Funnel](./resources/code_funnel.png)
 
-### **`drivebase`**
-This is responsible for implementing *how* a drivebase should work depending on user input.
+## Mission Control
+The mission control is a piece of software that allows us to interact with the rover over a browser as it is written in Vue.js/Typescript. 
+Cameras, controller input, telemetry data, controlling motors, life-detection results, and so on are all shown here. 
 
-Through a subscriber, it will recieve signals to when execute a high level control scheme: "move left side", "move right side", "move forwards", etc.
+A very important concept is that the mission control acts as a **CLIENT** when sending and getting data from the rover. This means that the mission control
+should never represent the current state of the rover; i.e. you cannot represent what mode the arm is in (individual motor control, disabled, inverse kinematics, etc)
+in the mission control 
```

Comment:
perhaps change the last line from `i.e. you cannot...` to `mission control reads data from the rover, not the other way around.` or something along those lines